### PR TITLE
Issue 141 claim to backstop

### PR DESCRIPTION
**Pull request by markuspluna**
User claims now are deposited directly into whatever backstop module they previously deposited in

### CODE REVIEW

File: backstop/src/emissions/claim.rs [hunk 1]
```diff
@@ -13,26 +14,63 @@ pub fn execute_claim(e: &Env, from: &Address, pool_addresses: &Vec<Address>, to:
...
+            let to_mint = pool_balance.convert_to_shares(deposit_amount);
+            pool_balance.deposit(deposit_amount, to_mint);
+            user_balance.add_shares(to_mint);
+
+            storage::set_pool_balance(e, &pool_id, &pool_balance);
+            storage::set_user_balance(e, &pool_id, to, &user_balance);
+            e.events().publish(
+                (Symbol::new(&e, "deposit"), pool_id, to),
+                (deposit_amount, to_mint),
+            );
```
Comment (mootz12) [inline backstop/src/emissions/claim.rs:60]: not a massive fan of this logic being duplicated, but there doesn't seem to be that easy of a way around it without a bunch of refactoring to make things consistent

File: backstop/src/emissions/claim.rs [hunk 2]
```diff
@@ -13,26 +14,63 @@ pub fn execute_claim(e: &Env, from: &Address, pool_addresses: &Vec<Address>, to:
...
-        let blnd_token = TokenClient::new(e, &storage::get_blnd_token(e));
-        blnd_token.transfer(&e.current_contract_address(), to, &claimed);
+        let blnd_id = storage::get_blnd_token(e);
+        let lp_id = storage::get_backstop_token(e);
+        TokenClient::new(e, &blnd_id).approve(
+            &e.current_contract_address(),
+            &lp_id,
+            &claimed,
+            &(e.ledger().sequence() + 10),
+        );
```
[the remark this answers is not in the document]
Comment (mootz12) [reply backstop/src/emissions/claim.rs:35]: This being said, this PR should probably follow the Comet one.

File: backstop/src/storage.rs [hunk 3]
```diff
@@ -41,8 +41,8 @@ pub struct UserEmissionData {
 #[derive(Clone)]
 #[contracttype]
 pub struct PoolUserKey {
-    pool: Address,
-    user: Address,
+    pub pool: Address,
+    pub user: Address,
```
Comment (mootz12) [inline backstop/src/storage.rs:45]: not sure I see why these needed to be public?

**Review by mootz12 — COMMENTED** [review]
Approving - but we should hold off merging for https://github.com/blend-capital/blend-contracts/pull/143

File: backstop/src/storage.rs (hunk 3 above)
Comment (markuspluna) [reply backstop/src/storage.rs:45]: I needed them public for tests and forgot to remove whoops

**Review by mootz12 — APPROVED** [review]
LG2M

**Pull request merged**